src/Stratis.Bitcoin.IntegrationTests/RPC/RPCTestsMutable.cs

```diff
@@ -29,5 +32,67 @@ public void TestRpcGetBlockWithValidHashIsSuccessful()
             }
         }
+
+        [Fact]
+        public void TestRpcSendToAddressIsSuccessful()
+        {
+            using (NodeBuilder builder = NodeBuilder.Create(this))
+            {
+                Network network = KnownNetworks.RegTest;
+                var node = builder.CreateStratisPowNode(network);
+                builder.StartAll();
+                RPCClient rpcClient = node.CreateRPCClient();
+                node.NotInIBD().WithWallet();
+                int blocksToMine = (int)network.Consensus.CoinbaseMaturity + 1;
+                TestHelper.MineBlocks(node, blocksToMine);
+                TestHelper.WaitLoop(() => node.FullNode.GetBlockStoreTip().Height == blocksToMine);
+
+                var alice = new Key().GetBitcoinSecret(network);
+                var aliceAddress = alice.GetAddress();
+                rpcClient.WalletPassphrase("password", 60);
+                var txid = rpcClient.SendToAddress(aliceAddress, Money.Coins(1.0m));
+                rpcClient.SendCommand(RPCOperations.walletlock); // TODO: create a method for walletlock.
+
+                // Check the hash calculated correctly.
+                var tx = rpcClient.GetRawTransaction(txid);
+                Assert.Equal(txid, tx.GetHash());
+
+                // Check the output is the right amount.
+                var coin = tx.Outputs.AsCoins().First(c => c.ScriptPubKey == aliceAddress.ScriptPubKey);
+                Assert.Equal(coin.Amount, Money.Coins(1.0m));
+            }        
+        }
+
+        [Fact]
+        public void TestRpcSendToAddressCantSpendWhenLocked()
+        {
+            using (NodeBuilder builder = NodeBuilder.Create(this))
+            {
+                Network network = KnownNetworks.RegTest;
+                var node = builder.CreateStratisPowNode(network);
+                builder.StartAll();
+                RPCClient rpcClient = node.CreateRPCClient();
+                node.NotInIBD().WithWallet();
+                int blocksToMine = (int)network.Consensus.CoinbaseMaturity + 1;
+                TestHelper.MineBlocks(node, blocksToMine);
+                TestHelper.WaitLoop(() => node.FullNode.GetBlockStoreTip().Height == blocksToMine);
+
+                var alice = new Key().GetBitcoinSecret(network);
+                var aliceAddress = alice.GetAddress();
+
+                // Not unlockedcase.                
+                Assert.Throws<RPCException>(() => rpcClient.SendToAddress(aliceAddress, Money.Coins(1.0m)));
+
+                // Unlock and lock case.
+                rpcClient.WalletPassphrase("password", 60);
+                rpcClient.SendCommand(RPCOperations.walletlock); // TODO: create a method for walletlock.
+                Assert.Throws<RPCException>(() => rpcClient.SendToAddress(aliceAddress, Money.Coins(1.0m)));
+
+                // Unlock timesout case.
+                rpcClient.WalletPassphrase("password", 5);
+                Thread.Sleep(120 * 1000); // 2 minutes.
```

Comment:
Maybe use a much shorter duration than this? It will just slow the tests down, we expect the wallet to be locked again within 5 seconds, yes?

Reply:
Ah, I see you mention this in the PR description.